R/retentionIndexing.R

```diff
@@ -0,0 +1,48 @@
+#' @title Convert retention times to retention indices
+#'
+#' @description
+#'
+#' `rtimeIndexing` uses a list of known substances to convert retention times to
+#'     retention indices. By default linear interpolation is performed.
+#'
+#' @param x `numeric` vector with retention times
+#' 
+#' @param y `data.frame`data.frame containing two columns, where the first 
+#'     holds the retention times of the indexing substancs and the second the
+#'    actual index value
+#' 
+#' @param FUN `function` function defining how the conversion is performed, 
+#'     default is linear interpolation
+#'
+#' @return `numeric` vector of same length as x with retention indices
+#'
+#' @author Michael Witting
+#'
+#' @export
+#'
+#' @examples
+#'
+#' rti <- data.frame(rtime = c(1,2,3),
+#' rindex = c(100,200,300))
+#' rtime <- c(1.5, 2.5)
+#' rtimeIndexing(rtime, rti)
+#' 
+rtimeIndexing <- function(x,
+                          y,
+                          FUN = rtiLinear) {
+  
+  unlist(lapply(x, FUN = FUN, rti = y))
```

Comment:
The `approx` function is vectorized. So you could just use:
```r
FUN <- match.fun(FUN)
do.call(FUN, list(x = x, rti = y))
# or even FUN(x = x, y = rti)
```
If you think you really need a kind of *apply function here I would suggest `vapply`.

BTW: I would allow additional arguments to `FUN` (`...`).

Reply:
Thank you for the suggestion. `approx` is only one of the functions I would like to implement, there are also cubic splines etc.